unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/RPCSocialApiBridge.cs

```diff
@@ -0,0 +1,301 @@
+                    // {
+                    //     case FriendshipEventResponse.BodyOneofCase.None: break;
+                    //     case FriendshipEventResponse.BodyOneofCase.Request: break;
+                    //     case FriendshipEventResponse.BodyOneofCase.Accept: break;
+                    //     case FriendshipEventResponse.BodyOneofCase.Reject: break;
+                    //     case FriendshipEventResponse.BodyOneofCase.Delete: break;
+                    //     case FriendshipEventResponse.BodyOneofCase.Cancel: break;
+                    //     default: throw new ArgumentOutOfRangeException();
+                    // }
+                }
+            }
+        }
+
+        public async UniTask RejectFriendship(string friendRequestId, CancellationToken cancellationToken = default)
+        {
+            string userId = GetUserIdFromFriendRequestId(friendRequestId);
+
+            var updateFriendshipPayload = new UpdateFriendshipPayload()
+            {
+                Event =
+                    new FriendshipEventPayload()
+                    {
+                        Reject = new RejectPayload()
+                        {
+                            User = new User() { Address = userId }
+                        }
+                    }
+            };
+
+            await this.UpdateFriendship(updateFriendshipPayload, cancellationToken);
+
+            OnFriendRequestRemoved?.Invoke(userId);
+        }
+
+        public async UniTask<FriendRequest> RequestFriendship(string friendUserId, string messageBody, CancellationToken cancellationToken = default)
+        {
+            long createdAt = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
+
+            var updateFriendshipPayload = new UpdateFriendshipPayload()
+            {
+                Event =
+                {
+                    Request = new RequestPayload()
+                    {
+                        Message = messageBody,
+                        User = new User() { Address = friendUserId },
+                    }
+                }
+            };
+
+            await UpdateFriendship(updateFriendshipPayload, cancellationToken);
+
+            return new FriendRequest(GetFriendRequestId(friendUserId, createdAt), createdAt, "GET OWN ADDRESS", friendUserId, messageBody);
+        }
+
+        private static string GetUserIdFromFriendRequestId(string friendRequestId) =>
+            friendRequestId.Split("-")[0];
+
+        private static string GetFriendRequestId(string userId, long createdAt) =>
+            $"{userId}-{createdAt}";
+
+        public async UniTask UpdateFriendship(UpdateFriendshipPayload updateFriendshipPayload, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                // TODO: pass cancellation token to rpc client when is supported
+                var response = await socialClient
```

Comment:
`cancellationToken.ThrowIfCancellationRequested();` should be added after the response, since we cannot pass a cancellationToken to the rpcClient